exporters/otlp/lib/opentelemetry/exporters/otlp/exporter.rb

```diff
@@ -0,0 +1,252 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'opentelemetry/sdk'
+require 'net/http'
+
+require 'opentelemetry/proto/common/v1/common_pb'
+require 'opentelemetry/proto/resource/v1/resource_pb'
+require 'opentelemetry/proto/trace/v1/trace_pb'
+require 'opentelemetry/proto/collector/trace/v1/trace_service_pb'
+
+module OpenTelemetry
+  module Exporters
+    module OTLP
+      # An OpenTelemetry trace exporter that sends spans over HTTP as Protobuf encoded OTLP ExportTraceServiceRequests.
+      class Exporter
+        SUCCESS = OpenTelemetry::SDK::Trace::Export::SUCCESS
+        FAILURE = OpenTelemetry::SDK::Trace::Export::FAILURE
+        private_constant(:SUCCESS, :FAILURE)
+
+        # Default timeouts in seconds.
+        KEEP_ALIVE_TIMEOUT = 30
+        OPEN_TIMEOUT = 5
+        READ_TIMEOUT = 5
+        RETRY_COUNT = 5
+        PATH = '/v1/trace'
+        private_constant(:KEEP_ALIVE_TIMEOUT, :OPEN_TIMEOUT, :READ_TIMEOUT, :RETRY_COUNT, :PATH)
+
+        def initialize(host:,
+                       port:,
+                       path: PATH,
+                       use_ssl: false,
+                       keep_alive_timeout: KEEP_ALIVE_TIMEOUT,
+                       open_timeout: OPEN_TIMEOUT,
+                       read_timeout: READ_TIMEOUT,
+                       retry_count: RETRY_COUNT)
+          @http = Net::HTTP.new(host, port)
+          @http.use_ssl = use_ssl
+          @http.keep_alive_timeout = keep_alive_timeout
+          @http.open_timeout = open_timeout
+          @http.read_timeout = read_timeout
+
+          @path = path
+          @max_retry_count = retry_count
+          @tracer = OpenTelemetry.tracer_provider.tracer
+
+          @shutdown = false
+        end
+
+        # Called to export sampled {OpenTelemetry::SDK::Trace::SpanData} structs.
+        #
+        # @param [Enumerable<OpenTelemetry::SDK::Trace::SpanData>] span_data the
+        #   list of recorded {OpenTelemetry::SDK::Trace::SpanData} structs to be
+        #   exported.
+        # @return [Integer] the result of the export.
+        def export(span_data)
+          return FAILURE if @shutdown
+
+          send_bytes(encode(span_data))
+        end
+
+        # Called when {OpenTelemetry::SDK::Trace::Tracer#shutdown} is called, if
+        # this exporter is registered to a {OpenTelemetry::SDK::Trace::Tracer}
+        # object.
+        def shutdown
+          @shutdown = true
+          @http.finish if @http.started?
+        end
+
+        private
+
+        def send_bytes(bytes)
+          retry_count = 0
+          untraced do
```

Comment:
`Net::HTTP` requests in this block should be untraced.

Reply:
this is very nifty!